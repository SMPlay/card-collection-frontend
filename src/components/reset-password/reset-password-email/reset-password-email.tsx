import React from "react";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";
import { Button, TextField } from "@material-ui/core";

import { FormikType } from "../../../types/FormikType";
import { checkErrorByField } from "../../../utils";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
      "& .MuiTextField-root": {
        margin: theme.spacing(1),
        width: "60ch",
      },
    },
  })
);

export interface Values {
  email?: string;
}

export const ResetPasswordEmail: React.FC<FormikType<Values>> = ({
  handleSubmit,
  handleChange,
  values,
  status,
  errors,
}) => {
  const styles = useStyles();

  return (
    <form onSubmit={handleSubmit} className={styles.root}>
      <TextField
        id="email"
        value={values.email}
        onChange={handleChange}
        label="Email"
        placeholder="Введите email"
        variant="outlined"
        error={
          errors.email ||
          status?.error === "Пользователя с таким email не существует"
            ? true
            : false
        }
        helperText={
          checkErrorByField("resetEmail", status?.error)
            ? status?.error
            : errors.email
        }
        required
        disabled={status?.loading}
      />
      <Button
        variant="contained"
        color="primary"
        type="submit"
        disabled={status?.loading}
      >
        Отправить сообщение об восстановлении
      </Button>
    </form>
  );
};
