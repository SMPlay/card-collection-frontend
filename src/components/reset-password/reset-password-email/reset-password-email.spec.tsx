import React from "react";
import { create } from "react-test-renderer";

import { ResetPasswordEmail } from "./reset-password-email";

describe("ResetPasswordEmail compoennt", () => {
  it("should render", () => {
    const tree = create(<ResetPasswordEmail/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
