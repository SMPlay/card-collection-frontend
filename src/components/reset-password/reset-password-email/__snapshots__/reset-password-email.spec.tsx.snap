// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordEmail compoennt should render 1`] = `
<form
  className="makeStyles-root-1"
  onSubmit={[MockFunction]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
      data-shrink={true}
      htmlFor="email"
      id="email-label"
    >
      Email
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="email"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Введите email"
        required={true}
        type="text"
        value="test"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-4 PrivateNotchedOutline-legendNotched-5"
        >
          <span>
            Email
             *
          </span>
        </legend>
      </fieldset>
    </div>
    
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Отправить сообщение об восстановлении
    </span>
  </button>
</form>
`;
